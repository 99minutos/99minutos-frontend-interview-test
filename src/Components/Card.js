import React from 'react'

const Card = ({photo, name, detail, date, video}) => {
    return (
      <div className="w-full rounded overflow-hidden shadow-lg my-2">
        {photo!=null?
         <div className="flex justify-center">
          <img className="h-60 max-h-full md:max-h-screen bg-auto bg-no-repeat bg-center" src={photo} alt="Sunset in the mountains"/>
         </div>
        : null}
      <div className="px-6 py-4">
       <div className="font-bold text-xl mb-2">{name}</div>
       <p className="text-grey-darker text-base">
        {detail}
       </p>
       <div className="flex flex-row-reverse">
        {date!=null?
         <p className="text-grey-darker text-base">
           {date}
         </p>
        : null}
       </div>
       {video!=null?
        <div className="flex justify-center">
          <a
           href={video}
           rel="noopener noreferrer"
           target="_blank"
           aria-describedby="link-new-window"
          >
           <button className="btn btn-blue btn-blue:hover">See more</button>
           </a>
        </div>
        : null}
      </div>
    </div>
    )
}

export default Card
