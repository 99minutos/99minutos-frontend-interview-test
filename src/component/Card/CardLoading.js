import { Card, CardContent, Typography } from '@material-ui/core';
import { Skeleton } from '@material-ui/lab';
import React from 'react';
import './index.scss';

const CardLoading = () => (
    <Card className="card">
        <Skeleton variant="rect" width="35%">
            <div style={{ paddingTop: '57%' }} />
        </Skeleton>
        <div className="card-details">
            <CardContent>
                <Typography component="div" variant="h5">
                    <Skeleton width={'50%'}/>
                </Typography>
                <Typography component="div" variant="caption">
                    <Skeleton/>
                </Typography>
                <Typography component="div" variant="caption">
                    <Skeleton />
                </Typography>
                <Typography component="div" variant="caption">
                    <Skeleton />
                </Typography>
                <p className="date"><Skeleton width={100}/></p>
            </CardContent>
        </div>
    </Card>
    );

export default CardLoading;