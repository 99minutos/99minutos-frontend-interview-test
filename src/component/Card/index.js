import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Avatar, Card, CardContent, CardMedia, makeStyles } from '@material-ui/core';
import spacecraft from '../../images/spacecraft.png';
import _ from 'lodash';
import './index.scss';
import moment from 'moment';
import { SpaceContext } from '../../context/SpaceProvider';

const useStyles = makeStyles((theme) => ({
    large: {
      width: theme.spacing(8),
      height: theme.spacing(8),
      marginLeft: theme.spacing(2),
    },
  }));

const CardSpace = ({ space = {}, onClick }) =>{
    const classes = useStyles();
    const { spaceData } = useContext(SpaceContext);
    const { selected } = spaceData;
    return (
        <Card className={`card ${selected?.id === space?.id ? 'active': ''}`} onClick={onClick}>
            <CardMedia className="img-container">
                <Avatar src={!_.isEmpty(space?.links?.flickr_images) ? space?.links?.flickr_images[0] : spacecraft} alt={space?.mission_name ? space?.mission_name : 'default image'} className={classes.large}/>
            </CardMedia>
            <div className="card-details">
                <CardContent>
                    {
                        !_.isEmpty(space) ? (
                            <>
                                <h5 className="title">{space?.mission_name}</h5>
                                <p className="description">{space?.details ? space?.details : 'No Data Available'}</p>
                                <p className="date">{space?.launch_date_local ? moment(space?.launch_date_local).format('DD/MM/YYYY') : 'No Data Available'}</p>
                            </>
                        ): (
                            <h3 className="text-center">No Data Available</h3>
                        )
                    }
                </CardContent>
            </div>
        </Card>
    );
}

CardSpace.propTypes = {
    space: PropTypes.object.isRequired
}

export default CardSpace;