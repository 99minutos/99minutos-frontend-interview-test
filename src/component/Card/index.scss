
.card{
    display: flex;
    min-height: 150px;
    margin: 0.5rem 1rem;
    transition: 0.3s ease-in !important;
    cursor: pointer;
    &:hover{
        background-color: #34495e;
        color: white;
    }
    .img-container{
        margin: auto;
        .card-img{
            max-width: 80px;
        }
    }
    .card-details{
        display: flex;
        flex-direction: column;
        width: 100%;
        position: relative;
        .title{
            text-align: left;
            margin: 0;
            font-size: 12pt;
        }
        .description{
            text-align: left;
            margin: 1rem 0rem;
            max-height: 40px;
            text-overflow: ellipsis;
            overflow: hidden;
        }
        .date{
            text-align: right;
            position: absolute;
            bottom: 0;
            right: 0;
            margin-right: 1rem;
        }
    }
}
.active{
    background-color: #34495e !important;
    color: white !important;
}