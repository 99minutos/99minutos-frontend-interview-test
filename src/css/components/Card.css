.Card {
  padding: 10px;
  border: 2px solid var(--color-white-primary);
  margin-bottom: 20px;
  overflow: hidden;
  cursor: pointer;
}

.Card h2 {
  margin-bottom: 10px;
}

.Card p {
  margin-bottom: 10px;
}

.Card em {
  float: right;
  font-style: italic;
}