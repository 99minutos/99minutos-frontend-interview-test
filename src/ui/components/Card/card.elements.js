import styled from "styled-components";

export const CardContainer = styled.div`
  background-color: var(--card-color);
  padding: 1.5rem;
  border-radius: var(--card-radius);
  box-shadow: 0 4px 144px rgba(0, 0, 0, 0.04);
  cursor: pointer;
  transition: 0.3s;
  margin-bottom: 1rem !important;
  color: #000 !important;

  .cardDetails {
    text-align: justify;
  }
  .date {
    text-align: right;
    font-size: 1.2rem;
    font-weight: 500;
  }

  &:hover {
    transform: scale(1.05);
  }

  &:last-child {
    margin-bottom: 0;
  }
`;
