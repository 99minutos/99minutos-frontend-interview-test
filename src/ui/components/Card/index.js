import React from "react";

// Styles
import { CardContainer } from "./card.elements.js";

const Card = ({ launch }) => {
  return (
    <CardContainer>
      <h2>{launch.mission_name}</h2>
      <br />
      <p className="cardDetails">{launch.details || "No details :("}</p>
      <br />
      <p className="date">
        {new Date(launch.launch_date_utc).toLocaleDateString()}
      </p>
    </CardContainer>
  );
};

export default Card;
