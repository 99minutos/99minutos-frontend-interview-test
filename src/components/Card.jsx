import React from 'react';
import '../css/components/Card.css';
import format from 'date-fns/format';
import parseISO from 'date-fns/parseISO';

const Card = ({ handleClick, missionName, description, launchDateLocal }) => {
  return (
    <div onClick={handleClick} className="Card">
      <h2>{missionName}</h2>
      <p>{description || 'Omitted'}</p>
      <em>{format(parseISO(launchDateLocal), 'dd/MM/yyyy')}</em>
    </div>
  );
};

export default Card;