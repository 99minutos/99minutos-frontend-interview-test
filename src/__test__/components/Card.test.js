import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Card from '../../components/Card';

describe('<Card />', () => {
  const card = shallow(<Card missionName="International station" description="details" launchDateLocal="2015-04-27T19:03:00-04:00" />);
  test('Component Card render', () => {
    expect(card.length).toEqual(1);
  });

  test('It received the props', () => {
    expect(card.find('h2').text()).toBe("International station");
  });
});

describe('Card Snapshot', () => {
  test('Check UI of component Card', () => {
    const card = create(<Card missionName="International station" description="details" launchDateLocal="2015-04-27T19:03:00-04:00" />);
    expect(card.toJSON()).toMatchSnapshot();
  });
});