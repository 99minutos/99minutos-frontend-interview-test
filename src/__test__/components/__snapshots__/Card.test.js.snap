// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshot Check UI of component Card 1`] = `
<div
  className="Card"
>
  <h2>
    International station
  </h2>
  <p>
    details
  </p>
  <em>
    27/04/2015
  </em>
</div>
`;
